security/pkg/credentialfetcher/fetcher_test.go

```diff
@@ -62,6 +62,7 @@ func TestNewCredFetcher(t *testing.T) {
 	for id, tc := range testCases {
 		cf, err := NewCredFetcher(
 			tc.fetcherType, tc.trustdomain, tc.jwtPath, tc.identityProvider)
+		defer cf.Stop()
```

Comment:
it doesn't really matter here, but you are not stopping until ALL subtests complete. I suggest using `t.Run(id, func() { .. })`

Reply:
I have moved the test body into t.Run() now.